server/proxy/bindings.go

```diff
@@ -58,7 +60,10 @@ func (p *Proxy) GetBindings(r *incoming.Request, cc apps.Context) ([]apps.Bindin
 
 	for i := range allApps {
 		go func(app apps.App) {
-			apprequest := r.WithDestination(app.AppID)
+			ctx, done := context.WithTimeout(r.Ctx(), bindingsFetchTimeout)
+			defer done()
```

Comment:
> A lot of requests are happening, and they should not run for long. Maybe 10 seconds is a good midleground?

@hanzei I still think we should be defensive against one app taking longer than the rest of the apps. If one app takes forever, we should still show the bindings we fetched from the other apps.

In this case should we use a separate `ctx` per app? I genuinely don't know if it's necessary

Reply:
>  I still think we should be defensive against one app taking longer than the rest of the apps. If one app takes forever, we should still show the bindings we fetched from the other apps.

Agreed. The code currently does guard against that. Even with a shared `ctx` it would work that way. Given that we create a new `incoming.Request` anyway, we are fine with sharing a `ctx`.